src-java/connecteddevices-topology/connecteddevices-storm-topology/src/main/java/org/openkilda/wfm/topology/connecteddevices/service/PacketService.java

```diff
@@ -366,55 +401,49 @@ private Flow getFlowBySwitchIdInPortAndOutVlan(SwitchId switchId, int inPort, in
         }
     }
 
-    private SwitchConnectedDevice getOrCreateLldpDevice(LldpInfoData data, int vlan) {
-        Optional<SwitchConnectedDevice> device = switchConnectedDeviceRepository
-                .findLldpByUniqueFieldCombination(
-                        data.getSwitchId(), data.getPortNumber(), vlan, data.getMacAddress(),
-                        data.getChassisId(), data.getPortId());
-
-        if (device.isPresent()) {
-            return device.get();
-        }
-
-        Optional<Switch> sw = switchRepository.findById(data.getSwitchId());
+    private void createLldpDevice(LldpInfoData data, FlowRelatedData flowRelatedData) {
+        Optional<Switch> optionalSwitch = switchRepository.findById(data.getSwitchId());
+        int vlan = flowRelatedData.getOriginalVlan();
 
-        if (!sw.isPresent()) {
+        if (!optionalSwitch.isPresent()) {
```

Comment:
I'd pull this up so that `create...` method has only one responsibility: to create. And if it's not possible to create a device, then don't call create method. But maybe this is too much for this PR.

Reply:
You are right about responsibilities. But I`ve tried to make minimum logic changes and yes such refactoring could cause many changes. So, I would prefer to leave it in its current state.